tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -3299,6 +3319,93 @@ def explain_classification_errors(self, data, model = None, print_rules: bool =
         cls, columns = imodels.explain_classification_errors(data, predictions, labels, print_rules=print_rules)
         return cls
 
+    # TODO: Add .delete() method to easily clean-up clones?
+    #  Would need to be careful that user doesn't delete important things accidentally.
+    # TODO: Add .save_zip() and load_zip() methods to pack and unpack artifacts into a single file to simplify deployment code?
+    def clone(self,
+              path: str,
+              *,
+              return_clone: bool = False,
+              dirs_exist_ok: bool = False):
+        """
+        Clone the predictor and all of its artifacts to a new location on local disk.
+        This is ideal for use-cases where saving a snapshot of the predictor is desired before performing
+        more advanced operations (such as fit_extra and refit_full).
+
+        Parameters
+        ----------
+        path : str
+            Directory path the cloned predictor will be saved to.
+        return_clone : bool, default = False
+            If True, returns the loaded cloned TabularPredictor object.
+            If False, returns the local path to the cloned TabularPredictor object.
+        dirs_exist_ok : bool, default = False
+            If True, will clone the predictor even if the path directory already exists, potentially overwriting unrelated files.
+            If False, will raise an exception if the path directory already exists and avoid performing the copy.
+
+        Returns
+        -------
+        If return_clone == True, returns the loaded cloned TabularPredictor object.
+        If return_clone == False, returns the local path to the cloned TabularPredictor object.
+
+        """
+        assert path != self.path, f"Cannot clone into the same directory as the original predictor! (path='{path}')"
+        path_clone = shutil.copytree(src=self.path, dst=path, dirs_exist_ok=dirs_exist_ok)
+        logger.log(30, f"Cloned {self.__class__.__name__} located in '{self.path}' to '{path_clone}'.\n"
+                       f"\tTo load the cloned predictor: predictor_clone = {self.__class__.__name__}.load(path=\"{path_clone}\")")
+        if return_clone:
+            return self.__class__.load(path=path_clone)
+        else:
+            return path_clone
+
+    def clone_for_deployment(self,
```

Comment:
Can we merge this method in `.clone()`?

Reply:
This is possible, I don't think there is much objective difference between having it in 1 vs 2 methods.

The main motivation for having it separate is so that when a user is typing `predictor.c` they see both `clone` and `clone_for_deployment` which may inspire them to look up what the differences between the two are. Also, deployment optimized may be multiple steps in future, and it isn't easy to default all of them without it being clunky in 1 method. Happy to change to one method if we think it is better though. We can always deprecate the clone_for_deployment method and add in new args to clone.